src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Input/Stylus/Common/StylusPointCollection.cs

```diff
@@ -196,16 +196,14 @@ internal StylusPointCollection(StylusPointDescription stylusPointDescription, in
                 {
                     //copy the rest of the data
                     data = new int[dataLength];
-                    for (int localIndex = 0, rawArrayIndex = i + startIndex; localIndex < data.Length; localIndex++, rawArrayIndex++)
-                    {
-                        data[localIndex] = rawPacketData[rawArrayIndex];
-                    }
+                    var rawArrayStartIndex = i + startIndex;
+                    Array.Copy(rawPacketData, rawArrayStartIndex, data, 0, dataLength);
```

Comment:
It would be a bit faster and, in my opinion, a bit cleaner if you used span:
```
var rawArrayStartIndex = i + startIndex;
data = rawPacketData.AsSpan(rawArrayStartIndex, dataLength).ToArray();
```

Here's the results I get with a modified version of your benchmark [here](https://github.com/lindexi/lindexi_gd/commit/07e92a85c104b760def5a87aac1c0e6c237311f5):
|      Method | start | length |      Mean |    Error |    StdDev |    Median |
|------------ |------ |------- |----------:|---------:|----------:|----------:|
|   CopyByFor |     0 |     10 |  15.64 ns | 0.214 ns |  0.178 ns |  15.61 ns |
| CopyByArray |     0 |     10 |  16.50 ns | 0.146 ns |  0.129 ns |  16.51 ns |
|  CopyBySpan |     0 |     10 |  12.25 ns | 0.229 ns |  0.191 ns |  12.19 ns |
|   CopyByFor |     0 |     20 |  24.91 ns | 0.205 ns |  0.192 ns |  24.88 ns |
| CopyByArray |     0 |     20 |  20.62 ns | 0.241 ns |  0.213 ns |  20.65 ns |
|  CopyBySpan |     0 |     20 |  15.78 ns | 0.299 ns |  0.280 ns |  15.65 ns |
|   CopyByFor |     0 |    100 | 104.31 ns | 4.836 ns | 14.258 ns | 113.63 ns |
| CopyByArray |     0 |    100 |  35.01 ns | 0.618 ns |  0.607 ns |  35.01 ns |
|  CopyBySpan |     0 |    100 |  31.50 ns | 0.409 ns |  0.362 ns |  31.51 ns |
|   CopyByFor |    10 |     10 |  14.10 ns | 0.761 ns |  2.231 ns |  15.54 ns |
| CopyByArray |    10 |     10 |  16.56 ns | 0.379 ns |  0.354 ns |  16.42 ns |
|  CopyBySpan |    10 |     10 |  13.00 ns | 0.127 ns |  0.106 ns |  13.00 ns |
|   CopyByFor |    10 |     20 |  25.03 ns | 1.144 ns |  3.094 ns |  26.34 ns |
| CopyByArray |    10 |     20 |  19.23 ns | 0.908 ns |  2.662 ns |  20.60 ns |
|  CopyBySpan |    10 |     20 |  14.12 ns | 0.776 ns |  2.288 ns |  15.86 ns |
|   CopyByFor |    10 |    100 |  99.44 ns | 4.250 ns | 12.532 ns |  90.83 ns |
| CopyByArray |    10 |    100 |  34.42 ns | 0.663 ns |  1.614 ns |  34.19 ns |
|  CopyBySpan |    10 |    100 |  31.72 ns | 0.522 ns |  0.488 ns |  31.84 ns |
|   CopyByFor |   100 |     10 |  15.78 ns | 0.157 ns |  0.147 ns |  15.73 ns |
| CopyByArray |   100 |     10 |  16.73 ns | 0.160 ns |  0.150 ns |  16.68 ns |
|  CopyBySpan |   100 |     10 |  12.07 ns | 0.132 ns |  0.123 ns |  12.03 ns |
|   CopyByFor |   100 |     20 |  23.31 ns | 1.207 ns |  3.559 ns |  25.97 ns |
| CopyByArray |   100 |     20 |  20.53 ns | 0.215 ns |  0.201 ns |  20.53 ns |
|  CopyBySpan |   100 |     20 |  16.04 ns | 0.268 ns |  0.251 ns |  15.99 ns |
|   CopyByFor |   100 |    100 |  98.36 ns | 4.251 ns | 12.535 ns |  90.02 ns |
| CopyByArray |   100 |    100 |  34.49 ns | 0.771 ns |  1.557 ns |  34.06 ns |
|  CopyBySpan |   100 |    100 |  30.96 ns | 0.516 ns |  0.483 ns |  30.95 ns |

Reply:
Thank you @ThomasGoulet73

Reply:
|      Method | start | length |      Mean |    Error |    StdDev |    Median |  Gen 0 | Allocated |
|------------ |------ |------- |----------:|---------:|----------:|----------:|-------:|----------:|
|   CopyByFor |     0 |     10 |  16.08 ns | 0.324 ns |  0.303 ns |  16.06 ns | 0.0153 |      64 B |
| CopyByArray |     0 |     10 |  16.06 ns | 0.406 ns |  0.528 ns |  16.06 ns | 0.0153 |      64 B |
|  CopyBySpan |     0 |     10 |  12.90 ns | 0.333 ns |  0.295 ns |  12.84 ns | 0.0153 |      64 B |
|   CopyByFor |     0 |     20 |  26.96 ns | 0.602 ns |  0.693 ns |  26.77 ns | 0.0249 |     104 B |
| CopyByArray |     0 |     20 |  20.97 ns | 0.502 ns |  0.766 ns |  20.83 ns | 0.0249 |     104 B |
|  CopyBySpan |     0 |     20 |  17.67 ns | 0.435 ns |  0.837 ns |  17.49 ns | 0.0249 |     104 B |
|   CopyByFor |     0 |    100 | 114.30 ns | 1.991 ns |  1.765 ns | 113.58 ns | 0.1013 |     424 B |
| CopyByArray |     0 |    100 |  52.52 ns | 0.815 ns |  0.722 ns |  52.50 ns | 0.1013 |     424 B |
|  CopyBySpan |     0 |    100 |  49.56 ns | 1.053 ns |  0.933 ns |  49.69 ns | 0.1013 |     424 B |
|   CopyByFor |    10 |     10 |  16.20 ns | 0.292 ns |  0.244 ns |  16.13 ns | 0.0153 |      64 B |
| CopyByArray |    10 |     10 |  17.02 ns | 0.430 ns |  1.176 ns |  16.64 ns | 0.0153 |      64 B |
|  CopyBySpan |    10 |     10 |  12.74 ns | 0.346 ns |  0.614 ns |  12.57 ns | 0.0153 |      64 B |
|   CopyByFor |    10 |     20 |  34.42 ns | 1.507 ns |  4.443 ns |  34.78 ns | 0.0249 |     104 B |
| CopyByArray |    10 |     20 |  29.02 ns | 0.641 ns |  1.654 ns |  28.42 ns | 0.0249 |     104 B |
|  CopyBySpan |    10 |     20 |  26.80 ns | 0.808 ns |  2.358 ns |  26.03 ns | 0.0249 |     104 B |
|   CopyByFor |    10 |    100 | 157.84 ns | 3.929 ns | 11.523 ns | 159.42 ns | 0.1013 |     424 B |
| CopyByArray |    10 |    100 |  87.48 ns | 1.828 ns |  4.272 ns |  87.01 ns | 0.1013 |     424 B |
|  CopyBySpan |    10 |    100 |  81.85 ns | 1.363 ns |  1.208 ns |  81.71 ns | 0.1013 |     424 B |
|   CopyByFor |   100 |     10 |  23.60 ns | 0.746 ns |  2.188 ns |  23.42 ns | 0.0153 |      64 B |
| CopyByArray |   100 |     10 |  20.87 ns | 0.504 ns |  0.620 ns |  20.79 ns | 0.0153 |      64 B |
|  CopyBySpan |   100 |     10 |  18.98 ns | 0.517 ns |  1.483 ns |  18.56 ns | 0.0153 |      64 B |
|   CopyByFor |   100 |     20 |  34.20 ns | 0.770 ns |  0.720 ns |  34.30 ns | 0.0249 |     104 B |
| CopyByArray |   100 |     20 |  28.25 ns | 0.492 ns |  0.436 ns |  28.25 ns | 0.0249 |     104 B |
|  CopyBySpan |   100 |     20 |  24.87 ns | 0.520 ns |  0.578 ns |  24.68 ns | 0.0249 |     104 B |
|   CopyByFor |   100 |    100 | 145.90 ns | 2.927 ns |  2.595 ns | 145.63 ns | 0.1013 |     424 B |
| CopyByArray |   100 |    100 |  91.07 ns | 2.504 ns |  7.303 ns |  89.06 ns | 0.1013 |     424 B |
|  CopyBySpan |   100 |    100 |  85.64 ns | 1.744 ns |  2.663 ns |  84.49 ns | 0.1013 |     424 B |